pandas/core/frame.py

```diff
@@ -2891,7 +2940,11 @@ def align(self, other, join='outer', axis=None, level=None, copy=True,
                                             broadcast_axis=broadcast_axis)
 
     @Appender(_shared_docs['reindex'] % _shared_doc_kwargs)
-    def reindex(self, index=None, columns=None, **kwargs):
+    def reindex(self, labels=None, index=None, columns=None, axis=0,
```

Comment:
it might be easier to simply default ``axis=None``, then in the code if you can set it to 0 where appropriate. This way you can detect whether its set.

Reply:
This messes with the default displayed in the method signature though (I believe that was @jorisvandenbossche's reason for wanting `axis=0` as the default?).

Reply:
Yeah, I am not too fixed on it. It is just a trade-off between having a more informative function signature (having the actual default value) vs being able to raise an error in the corner case of the user specifying `df.reindex/rename(columns=.., axis=0)`

Reply:
oh, you have to give that up. We don't do this anywhere else. Sure if you *can* preserve it would be great, but generally wouldn't work too hard.

Reply:
when I mean we don't do this anywhere else is that when we need to change an arg to see if we need to deprcate it, it should simply be ``None``. this case is a little different so it may be possible to preserve it.